knowledge_plugins.functions.Function: reorganise

Pamplemousse opened the pull request:
By extracting encoding / decoding logic into its own class.

ltfish commented on angr/knowledge_plugins/functions/function.py:
```diff
@@ -13,10 +13,10 @@
 import claripy
 
-from ...protos import function_pb2, primitives_pb2
+from .function_parser import FunctionParser
```
Import `...` first, `.` last.

ltfish commented:
CI failed. @Pamplemousse fix CI and I'll hit merge.

ltfish commented:
Linting is still failing.

Pamplemousse commented:
Corrected some linting issues that came from mis-distributing `import`s across files in the original commit.
Did some minor cleaning in the `FunctionParser` content, so I added it too.

ltfish commented:
Merged (with some changes to api-doc waiting to be done). Thanks!
